arch/arm64/core/prep_c.c

```diff
@@ -17,10 +17,10 @@
 #include <kernel_internal.h>
 #include <zephyr/linker/linker-defs.h>
 
-__weak void z_arm64_mm_init(bool is_primary_core) { }
-
 extern void z_arm64_mm_init(bool is_primary_core);
 
+__weak void z_arm64_mm_init(bool is_primary_core) { }
```

Comment:
Can I ask why you change this? Is it for compiler warnings or?

Reply:
This is moving the function declaration to be after the function prototype.

This might also show up in static code analysis or maybe also with e.g. gcc
compiler warnings.

best regards,

Florian La Roche